cpp/src/arrow/flight/client.cc

```diff
@@ -1198,6 +1227,12 @@ Status FlightClient::Authenticate(const FlightCallOptions& options,
   return impl_->Authenticate(options, std::move(auth_handler));
 }
 
+Status FlightClient::AuthenticateBasicToken(
```

Comment:
Could we make this `arrow::Result<std::pair<>>`?

Reply:
Yep, I will make this change.